core/src/com/unciv/ui/victoryscreen/VictoryScreen.kt

```diff
@@ -175,22 +178,84 @@ class VictoryScreen(val worldScreen: WorldScreen) : PickerScreen() {
+                demoLabel.add(rankLabel.name.toLabel()).row()
+                demoLabel.addSeparator().fillX()
+                demographicsTable.add(demoLabel)
+                for (category in RankingType.values()) {
+                    val headers = Table().apply { defaults().pad(5f) }
+                    headers.add(category.name.replace('_', ' ').toLabel()).row()
+                    headers.addSeparator().fillX()
+                    demographicsTable.add(headers)
+                    }
+            } else {
+                demographicsTable.row()
+                demographicsTable.add(rankLabel.name.toLabel())
+
+                for (category in RankingType.values()) {
+                    @Suppress("NON_EXHAUSTIVE_WHEN") // RankLabels.Demographic treated above
+                    when (rankLabel) {
+                        RankLabels.Rank -> { // finds the current player's rank in each category
+                            val rank = majorCivs.filter{ it.isAlive() }.sortedByDescending { it.getStatForRanking(category) }.indexOfFirst { it.civName == gameInfo.currentPlayerCiv.civName } + 1
+                            demographicsTable.add(rank.toLabel())
+                        }
+
+                        RankLabels.Value -> { // finds the current player's value in each category
+                            demographicsTable.add(getCivGroup(gameInfo.currentPlayerCiv, ": " + gameInfo.currentPlayerCiv.getStatForRanking(category).toString(), playerCivInfo)).fillX()
+                        }
+                      
+                        RankLabels.Best -> { // finds civ with the best value in each category
+                            val bestCiv = majorCivs.filter{ it.isAlive() }.maxByOrNull { it.getStatForRanking(category) }!!
+                            demographicsTable.add(getCivGroup(bestCiv, ": " + bestCiv.getStatForRanking(category).toString(), playerCivInfo)).fillX()
+                        }
+                      
+                        RankLabels.Average -> { // calculates the average value in each category
+                            var totalScore = 0
+                            for (civ in majorCivs.filter{ it.isAlive()} ) totalScore += civ.getStatForRanking(category)
+                            val averageScore = totalScore / (majorCivs.filter{ it.isAlive() }.lastIndex + 1)
+                            demographicsTable.add(averageScore.toLabel())
+                        }
+                      
+                        RankLabels.Worst -> { // finds civ with the worst value in each category
+                            val worstCiv = majorCivs.filter{ it.isAlive() }.minByOrNull { it.getStatForRanking(category) }!!
+                            demographicsTable.add(getCivGroup(worstCiv, ": " + worstCiv.getStatForRanking(category).toString(), playerCivInfo)).fillX()
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Such a stack of closing braces is a classic readability hindrance. In some projects / languages it's common to comment each with what it is closing - so far haven't seen that in Unciv, but if someone likes it I wouldn't veto. Just an idea.

A tricky idea to _really_ trim that down would be to add 'cell evaluator' lambdas to the RankLabels enum, makes the horizontal loop something akin to `for (category..) { render(category.evaluator(params)) }`, but that enum would then be the ugly part... Nah.

Reply:
One really simple way is to take the Value, Best and Worst and just one-line them - they don't actually need braces :)